syne_tune/optimizer/schedulers/multiobjective/linear_scalarizer.py

```diff
@@ -0,0 +1,160 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+import logging
+from collections.abc import Iterable
+from itertools import groupby
+from typing import Dict, Any, List, Union, Callable, Optional
+
+import numpy as np
+
+from syne_tune.backend.trial_status import Trial
+from syne_tune.optimizer.scheduler import TrialScheduler, TrialSuggestion
+from syne_tune.optimizer.schedulers.fifo import FIFOScheduler
+
+logger = logging.getLogger(__name__)
+
+
+def _all_equal(iterable: Iterable) -> bool:
+    """
+    Check if all elements of an iterable are the same
+    https://stackoverflow.com/questions/3844801/check-if-all-elements-in-a-list-are-identical
+    """
+    g = groupby(iterable)
+    return next(g, True) and not next(g, False)
+
+
+class LinearScalarizedScheduler(TrialScheduler):
+    """Scheduler with a Linear Scalarization of multiple objectives.
+    This method optimizes a single objective equal to the linear scalarization of given two objectives.
+    The scalarized single objective is named: 'scalarized_<metric1>_<metric2>_..._<metricN>'
+
+
+    :param base_scheduler: Class for the single-objective scheduler to be used on the scalarized objective.
+        It will be initialized inside this Scheduler
+    :param config_space: Configuration space for evaluation function
+    :param metric: Names of metrics to optimize
+    :param mode: Modes of metrics to optimize (min or max). All must be matching.
+    :param scalarization_weights: Weights used to scalarize objectives, if None an array of 1s is used
+    :param kwargs: Additional arguments to base_scheduler
+    """
+
+    scalarization_weights: np.ndarray
+    single_objective_metric: str
+    base_scheduler: TrialScheduler
+
+    def __init__(
+        self,
+        config_space: Dict[str, Any],
+        metric: List[str],
+        mode: Union[List[str], str] = "min",
+        scalarization_weights: Union[np.ndarray, List[float]] = None,
+        base_scheduler: Callable[[Any], TrialScheduler] = FIFOScheduler,
+        **kwargs,
+    ):
+        super(LinearScalarizedScheduler, self).__init__(config_space)
+        if scalarization_weights is None:
+            scalarization_weights = np.ones(shape=len(metric))
+        self.scalarization_weights = np.asarray(scalarization_weights)
+
+        self.metric = metric
+        self.mode = mode
+
+        self.single_objective_metric = f"scalarized_{'_'.join(metric)}"
+        assert len(mode) >= 1, "At least one mode must be provided"
```

Comment:
I'd instead insist on `len(metric) > 1`, otherwise this class should not be used.